lib/vanagon/component.rb

```diff
@@ -290,6 +290,10 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      { name => { 'version' => version, 'ref' => options[:ref] }.delete_if { |k, v| !v } }
```

Comment:
This is a simpler hash than the previous construction. There's no intermediate hash being allocated, and there's no conditional population. I'm not sure what magic you're referring to?